[DependencyInjection] Fix regression in ordering service locators by priority

longwave opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 7.0
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #57344
| License       | MIT

Fixes a regression in Symfony 7.0 where service locators are no longer ordered by priority.

longwave committed:
Allow service locators to be ordered by priority.

longwave commented:
I tried to rebase onto 7.0 but there is a conflict - if you want me to open a separate PR for 7.0 let me know.

nicolas-grekas commented:
I had a quick look at the git history: #42532 removed this ksort, but it's still in 5.4?
I'm missing something then - why introduced in 7.0 if 5.4 has the ksort?
Can you dig this question a bit?

oliverquynh committed:
Test convert CompletionInput into string

nicolas-grekas committed:
[Filesystem][Mime] Fix transient tests

longwave commented:
The break was introduced in Symfony 7.0 in #50578.

The test coverage for this was incorrectly switched to the wrong order in https://github.com/symfony/symfony/commit/cc4ef49d6cd30e2fa61e636ac9008d284bce3c00#diff-2d095e6b8882e9c562ea8541b1c0c995b8fd438946e73b4bd4a74f9895a47366L215-R215

Similar changes to IntegrationTest also show this regression: https://github.com/symfony/symfony/commit/cc4ef49d6cd30e2fa61e636ac9008d284bce3c00#diff-4440c0daf07aa75a97e4cf9f147391d566ebaa04fe772bdf1c896f92f6292f38

nicolas-grekas committed:
minor #57570 [Console] Test convert CompletionInput into string (seriquynh)

This PR was merged into the 5.4 branch.

Discussion
----------

[Console] Test convert CompletionInput into string

| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | N/A
| License       | MIT

CompletionInput can be converted into string but there are no tests for it. So, I a simple test here based on it's current behavior.

Commits
-------

0a7477d754 Test convert CompletionInput into string

nicolas-grekas committed:
minor #57575 [Filesystem][Mime] Fix transient tests (nicolas-grekas)

This PR was merged into the 5.4 branch.

Discussion
----------

[Filesystem][Mime] Fix transient tests

| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

HttpClient already uses this port so that when running concurrently, we get 404s for the logo.

Commits
-------

c3bf9a6074 [Filesystem][Mime] Fix transient tests

nicolas-grekas committed:
Merge branch '5.4' into 6.4

* 5.4:
  [Filesystem][Mime] Fix transient tests
  Test convert CompletionInput into string

nicolas-grekas committed:
Merge branch '6.4' into 7.0

* 6.4:
  [Filesystem][Mime] Fix transient tests
  Test convert CompletionInput into string

nicolas-grekas commented:
Thank you @longwave.
